manifests/vanilla/deploy-validation-webhook.sh

```diff
@@ -0,0 +1,115 @@
+    esac
+    shift
+done
+
+[ -z "${namespace}" ] && namespace=vmware-system-csi
+
+service=vsphere-webhook-svc
+secret=vsphere-webhook-certs
+
+
+if [ ! -x "$(command -v openssl)" ]; then
+    echo "openssl not found"
+    exit 1
+fi
+
+if [ ! -x "$(command -v kubectl)" ]; then
+    echo "kubectl not found"
+    exit 1
+fi
+
+tmpdir=$(mktemp -d)
+echo "creating certs in tmpdir ${tmpdir} "
+
+cat <<EOF >> "${tmpdir}"/server.conf
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+prompt = no
+[req_distinguished_name]
+CN = ${service}.${namespace}.svc
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage = clientAuth, serverAuth
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = ${service}
+DNS.2 = ${service}.${namespace}
+DNS.3 = ${service}.${namespace}.svc
+EOF
+
+# Generate the CA cert and private key
+openssl req -nodes -new -x509 -keyout "${tmpdir}"/ca.key -out "${tmpdir}"/ca.crt -subj "/CN=vSphere CSI Admission Controller Webhook CA"
+openssl genrsa -out "${tmpdir}"/webhook-server-tls.key 2048
+openssl req -new -key "${tmpdir}"/webhook-server-tls.key -subj "/CN=vsphere-webhook-svc.vmware-system-csi.svc" -config "${tmpdir}"/server.conf \
+  | openssl x509 -req -CA "${tmpdir}"/ca.crt -CAkey "${tmpdir}"/ca.key -CAcreateserial -out "${tmpdir}"/webhook-server-tls.crt -extensions v3_req -extfile "${tmpdir}"/server.conf
+
+cat <<eof >"${tmpdir}"/webhook.config
+[WebHookConfig]
+port = "8443"
+cert-file = "/run/secrets/tls/tls.crt"
+key-file = "/run/secrets/tls/tls.key"
+eof
+
+kubectl delete secret ${secret} --namespace "${namespace}" 2>/dev/null || true
+
+# create the secret with CA cert and server cert/key
+kubectl create secret generic "${secret}" \
+        --from-file=tls.key="${tmpdir}"/webhook-server-tls.key \
+        --from-file=tls.crt="${tmpdir}"/webhook-server-tls.crt \
+        --from-file=webhook.config="${tmpdir}"/webhook.config \
+        --dry-run=client -o yaml |
+    kubectl -n "${namespace}" apply -f -
+
+
+CA_BUNDLE="$(openssl base64 -A <"${tmpdir}/ca.crt")"
+# clean-up previously created service and validatingwebhookconfiguration. Ignore errors if not present.
+
+kubectl delete service vsphere-webhook-svc --namespace "${namespace}" 2>/dev/null || true
+kubectl delete validatingwebhookconfiguration.admissionregistration.k8s.io validation.csi.vsphere.vmware.com --namespace "${namespace}" 2>/dev/null || true
+kubectl delete serviceaccount vsphere-csi-webhook --namespace "${namespace}" 2>/dev/null || true
+kubectl delete clusterrole.rbac.authorization.k8s.io vsphere-csi-webhook-role 2>/dev/null || true
+kubectl delete clusterrolebinding.rbac.authorization.k8s.io vsphere-csi-webhook-role-binding --namespace "${namespace}" 2>/dev/null || true
+kubectl delete deployment vsphere-csi-webhook --namespace "${namespace}" 2>/dev/null || true
+
+# patch validatingwebhook.yaml with CA_BUNDLE and create service and validatingwebhookconfiguration
```

Comment:
you could also use the `https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/vanilla/validatingwebhook.yaml` to get the yaml file, this way, users need to download just the script instead of cloning the repo.

Reply:
we can not use URL with the sed command

```
sed: https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/vanilla/validatingwebhook.yaml: No such file or directory
```

Reply:
Could you try:
```
curl https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/master/manifests/vanilla/validatingwebhook.yaml | sed "s/caBundle: .*$/caBundle: ${CA_BUNDLE}/g" | kubectl apply -f -
```

Reply:
done this worked. We will need to make sure to correct the URL when we cut the tag release  instead of using the master branch path

for example
https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/v2.3.0-rc.1/manifests/vanilla/validatingwebhook.yaml